core/runtime/ops.rs

```diff
@@ -872,4 +881,35 @@ mod tests {
     )?;
     Ok(())
   }
+
+  #[op2(core, fast)]
+  pub fn op_state_rc(state: Rc<RefCell<OpState>>, value: u32) -> u32 {
+    let old_value: u32 = state.borrow_mut().take();
+    state.borrow_mut().put(value);
+    println!("{value} {old_value}");
+    old_value
+  }
+
+  #[op2(core, fast)]
+  pub fn op_state_ref(state: &OpState) -> u32 {
+    let old_value: &u32 = state.borrow();
+    *old_value
+  }
+
+  #[op2(core, fast)]
+  pub fn op_state_mut(state: &mut OpState, value: u32) {
+    *state.borrow_mut() = value;
+  }
+
+  #[tokio::test]
+  pub async fn test_op_state() -> Result<(), Box<dyn std::error::Error>> {
+    run_test2(
+      10000,
+      "op_state_rc",
+      "if (__index__ == 0) { op_state_rc(__index__) } else { assert(op_state_rc(__index__) == __index__ - 1) }",
```

Comment:
Where is that `__index__` coming from?

Reply:
This is provided by the `run_test2` method.